src/pds_doi_service/core/outputs/datacite/datacite_web_parser.py

```diff
@@ -55,87 +55,114 @@ def _parse_id(record):
             else:
                 # Parse the ID from the DOI field, it it's available
                 return record.get('doi').split('/')[-1]
-        except (AttributeError, KeyError) as err:
-            logger.warning('Could not parse id from record, reason: %s %s',
-                           err.__class__, err)
+        except (AttributeError, KeyError):
+            logger.warning('Could not parse optional field "id"')
```

Comment:
Felt it wasn't particularly useful information to have just to notify that we couldn't find an "optional" field to parse.  Including the nitty-gritty of whether it was because of a missing key, attribute, index-oob etc. felt to me like it was revealing unnecessary implementation details.